peer.go

```diff
@@ -2134,10 +2151,11 @@ func (p *peer) handleLocalCloseReq(req *htlcswitch.ChanClose) {
 	// out this channel on-chain, so we execute the cooperative channel
 	// closure workflow.
 	case htlcswitch.CloseRegular:
-		// First, we'll fetch a fresh delivery address that we'll use
-		// to send the funds to in the case of a successful
-		// negotiation.
-		deliveryAddr, err := p.genDeliveryScript()
+		// First, we'll fetch a  delivery address that we'll use to send the
+		// funds to in the case of a successful negotiation.
+		deliveryAddr, err := p.getLocalDeliveryScript(
```

Comment:
Not sure a method is needed here? Maybe it reads better just as 
```go
// First, we'll fetch a  delivery address that we'll use to send the
// funds to in the case of a successful negotiation.
deliveryAddr, err := channel.LocalUpfrontShutdownScript()

// If no upfromnt script was set, get anew address from the wallet.
if err == lnwallet.ErrNoLocalUpfrontShutdownScript:
        deliveryAddr, err = p.genDeliveryScript()
}
if err != nil {
...
```
alternatively match on the zero length script here, and fetch a new one in that case.